util/generate_xrc.py

```diff
@@ -0,0 +1,124 @@
+#!XRCED:begin-block:xrc%(windowName)s.PreCreate
+    def PreCreate(self, *args):
+        \"\"\" This function is called during the class's initialization.
+
+        Override it for custom setup before the window is created usually to
+        set additional window styles using SetWindowStyle() and SetExtraStyle().
+        \"\"\"
+        pass
+
+#!XRCED:end-block:xrc%(windowName)s.PreCreate
+
+    def __init__(self, parent):
+        if wx.MAJOR_VERSION == 3:
+            # Two stage creation (see http://wiki.wxpython.org/index.cgi/TwoStageCreation)
+            pre = wx.Pre%(windowClass)s()
+            self.PreCreate(pre)
+            get_resources().LoadOn%(windowClass)s(pre, parent, "%(windowName)s")
+            self.PostCreate(pre)
+        else:
+            wx.%(windowClass)s.__init__(self)
+            self.PreCreate()
+            get_resources().Load%(windowClass)s(self, parent, "%(windowName)s")
+
+        # Define variables for the controls, bind event handlers
+"""
+
+# Fix bytearray (on Python2?)
+PythonTemplates.ADD_FILE_TO_MEMFS = """\
+    wx.MemoryFSHandler.AddFile('XRC/%(memoryPath)s/%(filename)s', bytearray(%(filename)s))
+"""
+
+
+class OdemisXmlResourceCompiler(XmlResourceCompiler):
+
+    def NodeContainsFilename(self, node):
+        """ Does 'node' contain filename information at all? """
+
+        if node.nodeName == "icon_on":
+            return True
+            
+        return XmlResourceCompiler.NodeContainsFilename(self, node)
+
+    # Fixed version, for label of the menu
+    def GenerateWidgetClass(self, windowClass, windowName, topWindow, vars):
+        outputList = []
+
+        # output the header
+        outputList.append(self.templates.CLASS_HEADER % locals())
+
+        # Generate an attribute for each named item in the container
+        for widget in topWindow.getElementsByTagName("object"):
+            if not self.CheckAssignVar(widget): continue
+            widgetClass = widget.getAttribute("class")
+            widgetClass = re.sub("^wx", "", widgetClass)
+            widgetName = widget.getAttribute("name")
+            if widgetName != "" and widgetClass != "":
+                vars.append(widgetName)
+                if widgetClass == "MenuBar":
+                    outputList.append(self.templates.FRAME_MENUBAR_VAR % locals())
+                elif widgetClass == "MenuItem":
+                    outputList.append(self.templates.FRAME_MENUBAR_MENUITEM_VAR % locals())
+                elif widgetClass == "Menu":
+                    # Only look directly under for the "label"
+                    for e in widget.childNodes:
+                        if e.nodeType == e.ELEMENT_NODE and e.tagName == "label":
+                            label = e
+                            break
+                    label = label.childNodes[0].data
+                    outputList.append(self.templates.FRAME_MENUBAR_MENU_VAR % locals())
+#                 elif widgetClass == "ToolBar":
```

Comment:
Do you need the commented lines?

Reply:
Yes, because it's a (tweaked) copy of pywxrc, which has these lines.